core/sql/exp/exp_conv.cpp

```diff
@@ -1445,6 +1445,10 @@ ex_expr::exp_return_type convUnicodeToDatetime(char *target,
    }
 
    ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR);
+   char hexstr[256];
+   if(*diagsArea)
```

Comment:
Can heap, or diagsArea be null here (or other places below)? ExRaiseSqlError does allow these 2 possibilities and returns NULL in both cases. We may be able to convince ourselves that this should never happen, I get uneasy seeing a dereference of a pointer.
 One possibility is that we change 1447 to be 
ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR, NULL, NULL, NULL, NULL, stringToHex(hexstr, sizeof(hexstr) , source, sourceLen ));
Then lines 1449 and 1450 can be removed.